serviceorchestration-simple/src/main/java/eu/arrowhead/serviceorchestration/jpa/service/SimpleStoreDbService.java

```diff
@@ -0,0 +1,337 @@
+ *
+ * Contributors:
+ *  	AITIA - implementation
+ *  	Arrowhead Consortia - conceptualization
+ *
+ *******************************************************************************/
+package eu.arrowhead.serviceorchestration.jpa.service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import eu.arrowhead.common.service.util.ServiceInstanceIdUtils;
+import org.apache.commons.lang3.tuple.Triple;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.Assert;
+
+import eu.arrowhead.common.Utilities;
+import eu.arrowhead.common.exception.ArrowheadException;
+import eu.arrowhead.common.exception.InternalServerError;
+import eu.arrowhead.common.exception.InvalidParameterException;
+import eu.arrowhead.dto.OrchestrationSimpleStoreRequestDTO;
+import eu.arrowhead.serviceorchestration.jpa.entity.OrchestrationStore;
+import eu.arrowhead.serviceorchestration.jpa.repository.OrchestrationStoreRepository;
+
+import static org.hibernate.validator.internal.util.Contracts.assertTrue;
+
+@Service
+public class SimpleStoreDbService {
+
+	//=================================================================================================
+	// members
+
+	private final Logger logger = LogManager.getLogger(this.getClass());
+
+	@Autowired
+	private OrchestrationStoreRepository storeRepo;
+
+	private static final Object LOCK = new Object();
+
+	//=================================================================================================
+	// methods
+
+	//-------------------------------------------------------------------------------------------------
+	@Transactional(rollbackFor = ArrowheadException.class)
+	public List<OrchestrationStore> createBulk(final List<OrchestrationSimpleStoreRequestDTO> candidates, final String requesterName) {
+		logger.debug("createBulk started...");
+		Assert.isTrue(!Utilities.isEmpty(requesterName), "requesterName is empty");
+		Assert.isTrue(!Utilities.isEmpty(candidates), "candidate list is empty");
+		Assert.isTrue(!Utilities.containsNull(candidates), "candidate list contains null element");
+
+		final List<OrchestrationStore> toSave = candidates
+				.stream().map(n -> new OrchestrationStore(
+						n.consumer(),
+						ServiceInstanceIdUtils.retrieveServiceDefinitionFromInstanceId(n.serviceInstanceId()),
+						n.serviceInstanceId(),
+						n.priority(),
+						requesterName)).collect(Collectors.toList());
+
+		checkUniqueFields(candidates);
+
+		try {
+			return storeRepo.saveAllAndFlush(toSave);
+		} catch (final Exception ex) {
```

Comment:
still missing the InvalidParameterException catch branch in which you just rethrow the exception